components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -91,6 +99,74 @@ - (void)setSelectedItem:(UITabBarItem *)selectedItem {
   }
 
   _selectedItem = selectedItem;
+  UIView *itemView = self.itemViews[itemIndex];
+  if ([itemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+    MDCTabBarViewItemView *selectedItemView = (MDCTabBarViewItemView *)itemView;
+    selectedItemView.image = selectedItem.selectedImage ?: selectedItem.image;
+  }
+}
+
+#pragma mark - Key-Value Observing (KVO)
+
+- (void)addObserversToTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item addObserver:self
+           forKeyPath:kImageString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+    [item addObserver:self
+           forKeyPath:kSelectedImageString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+    [item addObserver:self
+           forKeyPath:kTitleString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+  }
+}
+
+- (void)removeObserversFromTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    @try {
+      [item removeObserver:self forKeyPath:kImageString];
+      [item removeObserver:self forKeyPath:kSelectedImageString];
+      [item removeObserver:self forKeyPath:kTitleString];
+    } @catch (NSException *exception) {
+      if (exception) {
+        // No need to do anything if there are no observers.
+      }
+    }
+  }
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (!context) {
```

Comment:
Please create a context variable and check it here, and add a call to super if the context does not match. See MDCButtonBar.m for the generally-recommended pattern.

Reply:
Done.

Reply:
Done.